modules/miner/blockmanager_test.go

```diff
@@ -1,115 +1,227 @@
 package miner
 
 import (
-	"errors"
+	"bytes"
 	"testing"
 
+	"github.com/NebulousLabs/Sia/crypto"
+	"github.com/NebulousLabs/Sia/modules"
 	"github.com/NebulousLabs/Sia/types"
 )
 
-// reconstructBlock reconstructs a block given the miner and the header
-func reconstructBlock(m *Miner, header types.BlockHeader) (*types.Block, error) {
-	block, exists := m.blockMem[header]
-	if !exists {
-		return nil, errors.New("Header is either invalid or too old")
+// solveHeader takes a block header as input and returns a solved block header
+// as output.
+func solveHeader(header types.BlockHeader, target types.Target) types.BlockHeader {
+	// Solve the header.
+	for {
+		// Incrememnt the nonce first to guarantee that a new header is formed
+		// - this helps check for pointer errors.
+		header.Nonce[0]++
```

Comment:
each attempt has 50% pass rate

256 attempts: 1-in-2^128 chance of failure.

Cryptographic grade probability that this test will never fail